src/plugins/dashboard/public/dashboard_listing/dashboard_listing.tsx

```diff
@@ -0,0 +1,264 @@
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import { FormattedRelative, I18nProvider } from '@kbn/i18n-react';
+import React, { PropsWithChildren, useCallback, useState } from 'react';
+
+import {
+  TableListView,
+  TableListViewKibanaDependencies,
+  TableListViewKibanaProvider,
+  type UserContentCommonSchema,
+} from '@kbn/content-management-table-list';
+import { ViewMode } from '@kbn/embeddable-plugin/public';
+import { reportPerformanceMetricEvent } from '@kbn/ebt-tools';
+import { toMountPoint, useExecutionContext } from '@kbn/kibana-react-plugin/public';
+import type { SavedObjectsFindOptionsReference, SimpleSavedObject } from '@kbn/core/public';
+
+import {
+  SAVED_OBJECT_DELETE_TIME,
+  SAVED_OBJECT_LOADED_TIME,
+  DASHBOARD_SAVED_OBJECT_TYPE,
+} from '../dashboard_constants';
+import {
+  dashboardListingTableStrings,
+  dashboardListingErrorStrings,
+} from './_dashboard_listing_strings';
+import { DashboardAttributes } from '../../common';
+import { pluginServices } from '../services/plugin_services';
+import { confirmCreateWithUnsaved } from './confirm_overlays';
+import { DashboardUnsavedListing } from './dashboard_unsaved_listing';
+import { DashboardApplicationService } from '../services/application/types';
+import { DashboardListingEmptyPrompt } from './dashboard_listing_empty_prompt';
+
+// because the type of `application.capabilities.advancedSettings` is so generic, the provider
+// requiring the `save` key to be part of it is causing type issues - so, creating a custom type
+type TableListViewApplicationService = DashboardApplicationService & {
+  capabilities: { advancedSettings: { save: boolean } };
+};
+
+const SAVED_OBJECTS_LIMIT_SETTING = 'savedObjects:listingLimit';
+const SAVED_OBJECTS_PER_PAGE_SETTING = 'savedObjects:perPage';
+
+interface DashboardSavedObjectUserContent extends UserContentCommonSchema {
+  attributes: {
+    title: string;
+    description?: string;
+    timeRestore: boolean;
+  };
+}
+
+const toTableListViewSavedObject = (
+  savedObject: SimpleSavedObject<DashboardAttributes>
+): DashboardSavedObjectUserContent => {
+  const { title, description, timeRestore } = savedObject.attributes;
+  return {
+    type: 'dashboard',
+    id: savedObject.id,
+    updatedAt: savedObject.updatedAt!,
+    references: savedObject.references,
+    attributes: {
+      title,
+      description,
+      timeRestore,
+    },
+  };
+};
+
+export type DashboardListingProps = PropsWithChildren<{
+  initialFilter?: string;
+  useSessionStorageIntegration?: boolean;
+  goToDashboard: (dashboardId?: string, viewMode?: ViewMode) => void;
+  getDashboardUrl: (dashboardId: string, usesTimeRestore: boolean) => string;
```

Comment:
From the [issue](https://github.com/elastic/kibana/issues/153869)

> An additional request for the dashboards list:
The tag is default Security Solution, and it cannot be unselected. Only dashboards tagged with Security Solution display here.

How could we support this for the consumer? Do we just have them set the `initialFilter` (i.e. `tag:(Security)`). I think that would filter the list correctly, but it can still be modified.

Reply:
Yes you're very correct that this PR doesn't solve all of the requests from the security team. I have removed the "closes" comment to leave the issue open after this one is merged.. we'll have a substantial amount of work to follow up on to cover that request fully.